crates/sdl-encoder/src/enum_def.rs

```diff
@@ -0,0 +1,143 @@
+use crate::EnumValue;
+use std::fmt::{self, Display};
+
+/// Enums are special scalars that can only have a defined set of values.
+///
+/// *EnumTypeDefinition*:
+///     Description<sub>opt</sub> **enum** Name Directives<sub>\[Const\] opt</sub> EnumValuesDefinition <sub>opt</sub>
+///
+/// Detailed documentation can be found in [GraphQL spec](https://spec.graphql.org/draft/#sec-Enums).
+///
+/// ### Example
+/// ```rust
+/// use sdl_encoder::{EnumValue, EnumDef};
+///
+/// let mut enum_ty_1 = EnumValue::new("CatTree".to_string());
+/// enum_ty_1.description(Some("Top bunk of a cat tree.".to_string()));
+/// let enum_ty_2 = EnumValue::new("Bed".to_string());
+/// let mut enum_ty_3 = EnumValue::new("CardboardBox".to_string());
+/// enum_ty_3.deprecated(Some("Box was recycled.".to_string()));
+///
+/// let mut enum_ = EnumDef::new("NapSpots".to_string());
+/// enum_.description(Some("Favourite cat nap spots.".to_string()));
+/// enum_.value(enum_ty_1);
+/// enum_.value(enum_ty_2);
+/// enum_.value(enum_ty_3);
+///
+/// assert_eq!(
+///     enum_.to_string(),
+///     r#""""Favourite cat nap spots."""
+/// enum NapSpots {
+///   """Top bunk of a cat tree."""
+///   CatTree
+///   Bed
+///   CardboardBox @deprecated(reason: "Box was recycled.")
```

Comment:
This is a small nit, but by common convention enum values in GraphQL are uppercased.